test/bases/renku_data_services/data_api/test_repositories.py

```diff
@@ -111,20 +114,20 @@ async def test_get_one_repository(oauth2_test_client: SanicASGITestClient, user_
     connection = await create_oauth2_connection("provider_1")
     repository_url = "https://example.org/username/my_repo.git"
 
-    _, res = await oauth2_test_client.get(f"/api/data/repositories/{quote_plus(repository_url)}", headers=user_headers)
+    _, res = await oauth2_test_client.get(
+        f"/api/data/repository?url={quote_plus(repository_url)}", headers=user_headers
+    )
 
     assert res.status_code == 200, res.text
     assert res.json is not None
     result = res.json
-    assert result.get("connection_id") == connection["id"]
-    assert result.get("provider_id") == "provider_1"
-    assert result.get("repository_metadata") is not None
-    repository_metadata = result["repository_metadata"]
-    assert repository_metadata.get("git_http_url") == repository_url
-    assert repository_metadata.get("permissions") is not None
-    permissions = repository_metadata["permissions"]
-    assert permissions.get("pull")
-    assert not permissions.get("push")
+    assert result["connection"]["id"] == connection["id"]
+    assert result["provider"]["id"] == "provider_1"
```

Comment:
You should use `.get` so that it's the `assert` which fails, not the get attribute on NoneType or similar if the response is malformed.

Reply:
Ok, thanks. I thought it doesn't matter what exactly fails in the tests. But I assume we have better error messages then?